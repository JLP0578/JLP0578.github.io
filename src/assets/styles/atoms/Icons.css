/***********************************/
/*             DEFAULT             */
/*            TELEPHONE            */
/***********************************/

span.icons.icons_high {
    color: #21e8cb;
}
span.icons.icons_neural {
    color: white;
}
span.icons.icons_low {
    color: #e68de1;
}
span.icons::before {
    content: '[';
    color: white;
}
span.icons::after {
    content: ']';
    color: white;
}