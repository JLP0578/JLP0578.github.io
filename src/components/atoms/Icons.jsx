import React from "react";
import PropTypes from 'prop-types';

require('../../assets/styles/atoms/Icons.css');

export default class Icons extends React.Component {

    addHigh(template) {
        return (
            <span className={"icons icons_"+template}>+</span>
        );
    }
    addNeutral(template) {
        return (
            <span className={"icons icons_"+template}>~</span>
        );
    }
    addLow(template) {
        return (
            <span className={"icons icons_"+template}>-</span>
        );
    }

    render() {
        const template = this.props.template || Icons.defaultProps.template;

        let result;
        switch (template) {
            case "high":
                result = this.addHigh(template);
                break;

            default:
            case "neutral":
                result = this.addNeutral(template);
                break;

            case "low":
                result = this.addLow(template);
                break;
        }

        return result;
    }

}

Icons.propTypes = {
    template: PropTypes.string,
};

Icons.defaultProps = {
    template: "neural",
};