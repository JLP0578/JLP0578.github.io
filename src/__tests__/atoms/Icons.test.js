import renderer from 'react-test-renderer';

import Icons from '../../components/atoms/Icons';

test('renders correctly', () => {
  const tree = renderer
    .create(<Icons />)
    .toJSON();

  expect(tree).toMatchSnapshot();

});