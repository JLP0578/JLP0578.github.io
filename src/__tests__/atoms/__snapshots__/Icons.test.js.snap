// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  className="icons icons_neural"
>
  ~
</span>
`;
